tutorials/data-science-preprocessing/index.md

```diff
@@ -0,0 +1,437 @@
+  libraries on your system, create a virtualenv and install the script
+  dependencies:
+
+      $ virtualenv venv
+      ...
+      $ source venv/bin/activate
+      (venv) $ pip install -r requirements.txt
+      ...
+      (venv) $
+
+With the BigQuery dataset created and your requirements installed, you are ready
+to run your pipeline.
+
+### Running locally
+
+    python clean.py meteors.json your-project-id:meteor_dataset.cleansed
+
+### Running on Cloud Dataflow
+
+To run the script on Cloud Dataflow, you'll have to specify your project, and a
+bucket in Google Cloud Storage:
+
+    PROJECT=your-project-id
+    # The $PROJECT.appspot.com bucket is created automatically with your project
+    BUCKET=gs://$PROJECT.appspot.com
+
+Now upload the json data file to Cloud Storage using the `gsutil` tool included
+in the Google Cloud SDK:
+
+    gsutil cp meteors.json gs://$BUCKET/
+
+Then run:
+
+    python clean.py gs://$BUCKET/meteors.json $PROJECT:meteor_dataset.cleansed \
+        --project $PROJECT \
+        --job_name $PROJECT-meteors \
+        --runner BlockingDataflowPipelineRunner \
+        --staging_location $BUCKET/staging \
+        --temp_location $BUCKET/temp \
+        --output $BUCKET/output
+
+You can check the status of your job on the [Dataflow
+dashboard](https://console.cloud.google.com/dataflow?project=_).
+
+## Sample query
+
+Now that the data has been ingested into BigQuery, it's available to explore.
+We'll go into more detail about this in the next section on [data
+exploration](exploration.md), but to give you an idea, here is a sample query we
+could do to find the number of meteor landings per year:
+
+     $ bq query "select year, count(*) from \
+         [$PROJECT:meteor_dataset.cleansed] \
+         group by year \
+         order by year desc \
+         limit 10"
+     Waiting on bqjob_r2a3ebeea2f95498c_00000158939ae926_1 ... (0s) Current
+     status: DONE
+     +------+------+
+     | year | f0_  |
+     +------+------+
+     | 2013 |    1 |
+     | 2012 |   15 |
+     | 2011 |  391 |
+     | 2010 |  420 |
+     | 2009 |  323 |
+     | 2008 |  261 |
+     | 2007 |  236 |
+     | 2006 | 1223 |
+     | 2005 |  246 |
+     | 2004 |  264 |
+     +------+------+
+
+## API Documentation
+
+We've only touched on a couple of the capabilities of Apache Beam and Cloud
+Dataflow. Take a look at the API documentation, and experiment with the other
+features.
+
+* Cloud Dataflow ([API Docs](/dataflow/docs))
+* Apache Beam ([Project page][beam] | [Python Docs][beam-py] | [Java Docs][beam-java])
+
+[beam-py]: https://github.com/apache/incubator-beam/tree/python-sdk/sdks/python
+[beam]: http://beam.incubator.apache.org/
+[beam-java]: http://beam.incubator.apache.org/documentation/programming-guide/
+
+## Cleanup
+
+To avoid recurring charges for resources created in this tutorial:
+
+* Delete the BigQuery table the data was outputted to. **Note**, though, that
```

Comment:
_Outputted_ isn't a word. I suggest, "Delete the BigQuery table that you used for your data output." Or words to that effect.